addons/sale_management/tests/test_sale_order.py

```diff
@@ -317,3 +317,26 @@ def test_option_creation(self):
             option.product_id = self.product_1
         order = order_form.save()
         self.assertTrue(bool(order.sale_order_option_ids.uom_id))
+
+    def test_option_price_unit_is_not_recomputed(self):
+        """
+        Verifies that user defined price unit for optional products remains the same after
+        update of quantities.
+        """
+
+        self.sale_order_option = self.env['sale.order.option'].create({
+            'name': "Option",
+            'product_id': self.optional_product.id,
+            'price_unit': 100,
+        })
+        self.sale_order_with_option = self.env['sale.order'].create({
+            'partner_id': self.partner.id,
+            'sale_order_option_ids': self.sale_order_option,
+        })
```

Comment:
```suggestion
        sale_order_with_option = self.env['sale.order'].create({
            'partner_id': self.partner.id,
            'sale_order_option_ids': [Command.create({
                'product_id': self.optional_product.id,
                'price_unit': 100.0,
            })],
        })
```

1) And there's no need to define a variable on self if it's not shared between execution scopes ;)
2) Create the option directly on the order, options are not meant to be created without an order (it's not required, but it should be someday :D)
3) why don't you create the option directly with the price_unit you need for your test (10) ?

Reply:
1,2 thank you :)

3) what I want to guaranteed is the price unit to remain the same after quantities have been changed, I'm emulating the flow:
 - add an option
 - change its price unit
 - add the optional product
 - change the quantities
 - the price unit should not be recomputed